listenbrainz/webserver/static/js/jsx/profile.jsx

```diff
@@ -77,17 +82,62 @@ class RecentListens extends React.Component {
     });
   }
 
-  sortListensByFollowUserRank(listens, userList){
-    if(userList.length <= 1){
-      return listens;
-    }
-    const currentListenIndex = listens.indexOf(this.state.currentListen);
-    let ignoredPastListens = [];
-    if(currentListenIndex !== -1){
-      ignoredPastListens = listens.splice(currentListenIndex);
+  setSortMethod(method){
+    if (["time","username","followList"].indexOf(method) === -1){
+      console.error("Trying to set sort method to unrecognized:",method);
+      return;
```

Comment:
"time" is the default set at line 34. I suppose I'm starting to be weary of all those strings instead of enums and guarding against a misuse.
A bit overkill TBH. Do you think I should remove those guards (this and next comment)?

Reply:
Yes, please do that. Then we'll merge and ask param to push it live.